### PR TITLE
[MXNET-1195] Cleanup Scala README file

### PR DESCRIPTION
**Pull request by piyushghai**
## Description ##
The scala-package/README file is outdated and contains some stale information. 
This PR aims to clean up the Scala README and also adds new information with regards to the inference APIs for Scala/Java as well as the new Memory Management system.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

@lanking520 @andrewfayres @zachgk

### CODE REVIEW

**Comment by piyushghai** [comment]
@mxnet-label-bot [Scala, Java, Documentation]

File: scala-package/README.md [hunk 1]
```diff
@@ -1,67 +1,119 @@
 <img src=https://raw.githubusercontent.com/dmlc/dmlc.github.io/master/img/logo-m/mxnet2.png width=135/> Deep Learning for Scala/Java
```
Comment (zachgk) [inline scala-package/README.md]: I don't think we want to keep the dmlc logo

File: scala-package/README.md [hunk 2]
```diff
@@ -1,67 +1,119 @@
...
 =====
 
-[![Build Status](http://jenkins.mxnet-ci.amazon-ml.com/job/incubator-mxnet/job/master/badge/icon)](http://jenkins.mxnet-ci.amazon-ml.com/job/incubator-mxnet/job/master/)
 [![GitHub license](http://dmlc.github.io/img/apache2.svg)](./LICENSE)
```
Comment (zachgk) [inline scala-package/README.md]: Also, I think we want to remove the license so that it inherits the license from the main mxnet project

File: scala-package/README.md [hunk 3]
```diff
@@ -1,67 +1,119 @@
...
+**macOS CPU**
+```HTML
+<dependency>
+  <groupId>org.apache.mxnet</groupId>
+  <artifactId>mxnet-full_2.11-osx-x86_64-cpu</artifactId>
+  <version><latest-mxnet-version></version>
+</dependency>
+```
+
+The ```<latest-mxnet-version>``` is a placeholder corresponding to the current MXNet version. The latest version information will be available on the [MXNet Maven Package Repository](https://search.maven.org/search?q=g:org.apache.mxnet).
```
Comment (zachgk) [inline scala-package/README.md]: Instead of using a placeholder, I think we should use something like "[1.4.0,)". This automatically picks up the latest version. We can also do "[1.4.0-SNAPSHOT,)" for nightly.

File: scala-package/README.md [hunk 4]
```diff
@@ -1,67 +1,119 @@
...
- 
-Technically, all you need is the `mxnet-full_2.11-{arch}-{xpu}-{version}.jar` in your classpath.
-It will automatically extract the native library to a tempfile and load it.
-You can find the pre-built jar file in [here](https://search.maven.org/search?q=g:org.apache.mxnet)
- and also our nightly build package [here](https://repository.apache.org/#nexus-search;gav~org.apache.mxnet~)
 
-Currently we provide `linux-x86_64-gpu`, `linux-x86_64-cpu` and `osx-x86_64-cpu`. Support for Windows will come soon.
-Use the following dependency in maven, change the artifactId according to your own architecture, e.g., `mxnet-full_2.11-osx-x86_64-cpu` for OSX (and cpu-only).
+The MXNet Scala/Java packages can be easily included in your Maven managed project.
+The stable jar files for the packages are available on the [MXNet Maven Package Repository](https://search.maven.org/search?q=g:org.apache.mxnet)
```
Comment (nswamy) [inline scala-package/README.md]: I think it might be better if you split into 
## Pre-Built Maven Packages
### Stable
### Nightly
Here you give the snippet to add the snapshot repo.
## Build From Source
Instead of search, can we add a badge link to the 3 types that we support and keep it consistent with the main page, the link here is more friendly than on the main page

File: scala-package/README.md [hunk 5]
```diff
@@ -1,67 +1,119 @@
...
+Currently we provide packages for Linux (CPU and GPU) and macOS (CPU only). Support for Windows will come soon.
```
Comment (nswamy) [inline scala-package/README.md]: Can you specify what Linux versions are support, if any required workarounds those as well(Ex: Centos).

File: scala-package/README.md (hunk 3 above)
Comment (nswamy) [reply scala-package/README.md]: Is this only for documentation purpose or does it work in the pom file?

File: scala-package/README.md [hunk 6]
```diff
@@ -70,123 +122,24 @@ make SCALA_TEST_ARGS=-Dsuites=org.apache.mxnet.NDArraySuite scalaunittest
...
-  org.apache.mxnet.examples.imclassification.TrainMnist \
-  --data-dir=./data/ \
-  --num-epochs=10 \
-  --network=mlp \
-  --cpus=0 \
-  --kv-store=dist_sync
-```
-
-Change the arguments and have fun!
+Examples & Usage
```
Comment (nswamy) [inline scala-package/README.md]: we shouldn't remove the example for training, can you please update the example?

Comment (piyushghai) [reply scala-package/README.md]: @nswamy I removed the example snippet to run here, as it seemed a bit out of place right under the section on build from source. 
I can however, direct the users to this script ```run_train_mnist.sh``` in order to run the training script for MNIST. 
Will that work ?

File: scala-package/README.md (hunk 5 above)
Comment (piyushghai) [reply scala-package/README.md]: Sure will do that.

File: scala-package/README.md (hunk 4 above)
Comment (piyushghai) [reply scala-package/README.md]: Sure. Will do that.

File: scala-package/README.md (hunk 6 above)
Comment (piyushghai) [reply scala-package/README.md]: Discussed offline, there would be a separate README for training code, elsewhere, and the main scala readme would point to it from here.

Comment (piyushghai) [reply scala-package/README.md]: I have added pointers to the training API usage in this commit : https://github.com/apache/incubator-mxnet/pull/13582/commits/91392c4d35a1f3781182ccab4d3ddd389647b9bf 

The improvement in usage of training APIs available on the website here : http://mxnet.incubator.apache.org/api/scala/model.html should be a part of separate task. 

Have added a JIRA item for it  : https://issues.apache.org/jira/browse/MXNET-1265
